demos/mask_rcnn_demo/cpp/main.cpp

```diff
@@ -45,240 +49,327 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
-
-                cv::Mat uchar_resized_mask(box_height, box_width, CV_8UC3,
-                    cv::Scalar(color.blue(), color.green(), color.red()));
-                roi_input_img.copyTo(uchar_resized_mask, resized_mask_mat <= MASK_THRESHOLD);
-
-                cv::addWeighted(uchar_resized_mask, alpha, roi_input_img, 1.0f - alpha, 0.0f, roi_input_img);
-                cv::rectangle(output_images[batch], roi, cv::Scalar(0, 0, 1), 1);
-            }
-        }
-        metrics.update(startTime);
-        for (size_t i = 0; i < output_images.size(); i++) {
-            std::string imgName = "out" + std::to_string(i) + ".png";
-            cv::imwrite(imgName, output_images[i]);
-            slog::info << "Image " << imgName << " created!" << slog::endl;
-        }
+        size_t class_id = static_cast<size_t>(box_info[1] + 1e-6);
 
-        slog::info << "Metrics report:" << slog::endl;
-        slog::info << "\tLatency: " << std::fixed << std::setprecision(1) << metrics.getTotal().latency << " ms" << slog::endl;
-    }
-    catch (const std::exception& error) {
-        slog::err << error.what() << slog::endl;
-        return 1;
+        if (prob > PROBABILITY_THRESHOLD && box_width > 0 && box_height > 0) {
+            size_t color_index = class_color.emplace(class_id, class_color.size()).first->second;
+            auto& color = CITYSCAPES_COLORS[color_index % arraySize(CITYSCAPES_COLORS)];
+            float* mask_arr = masks_data + box_stride * box + H * W * (class_id - 1);
+            slog::info << "Detected class " << class_id << " with probability " << prob << " from batch " << batch
+                       << ": [" << x1 << ", " << y1 << "], [" << x2 << ", " << y2 << "]" << slog::endl;
+            cv::Mat mask_mat(H, W, CV_32FC1, mask_arr);
+
+            cv::Rect roi = cv::Rect(static_cast<int>(x1), static_cast<int>(y1), box_width, box_height);
+            cv::Mat roi_input_img = output_images[batch](roi);
+            const float alpha = 0.7f;
+
+            cv::Mat resized_mask_mat(box_height, box_width, CV_32FC1);
+            cv::resize(mask_mat, resized_mask_mat, cv::Size(box_width, box_height));
+
+            cv::Mat uchar_resized_mask(box_height, box_width, CV_8UC3,
+                cv::Scalar(color.blue(), color.green(), color.red()));
+            roi_input_img.copyTo(uchar_resized_mask, resized_mask_mat <= MASK_THRESHOLD);
+
+            cv::addWeighted(uchar_resized_mask, alpha, roi_input_img, 1.0 - alpha, 0.0f, roi_input_img);
+            cv::rectangle(output_images[batch], roi, cv::Scalar(0, 0, 255), 1);
+        }
     }
-    catch (...) {
-        slog::err << "Unknown/internal exception happened." << slog::endl;
-        return 1;
+
+    metrics.update(startTime);
+
+    for (size_t i = 0; i < output_images.size(); i++) {
+        std::string imgName = "out" + std::to_string(i) + ".png";
+        cv::imwrite(imgName, output_images[i]);
```

Comment:
Could you also throw if `imwrite()` fails

Reply:
added imwrite return status check